src/Coders/Model/Factory.php

```diff
@@ -251,14 +251,79 @@ protected function prepareTemplate(Model $model, $name)
     protected function fillTemplate($template, Model $model)
     {
         $template = str_replace('{{namespace}}', $model->getBaseNamespace(), $template);
-        $template = str_replace('{{parent}}', $model->getParentClass(), $template);
-        $template = str_replace('{{properties}}', $this->properties($model), $template);
         $template = str_replace('{{class}}', $model->getClassName(), $template);
-        $template = str_replace('{{body}}', $this->body($model), $template);
+
+        $properties = $this->properties($model);
+        $usedClasses = $this->extractUsedClasses($properties);
+        $template = str_replace('{{properties}}', $properties, $template);
+
+        $parentClass = $model->getParentClass();
+        $usedClasses = array_merge($usedClasses, $this->extractUsedClasses($parentClass));
+        $template = str_replace('{{parent}}', $parentClass, $template);
+
+        $body = $this->body($model);
+        $usedClasses = array_merge($usedClasses, $this->extractUsedClasses($body));
+        $template = str_replace('{{body}}', $body, $template);
+
+        $usedClasses = array_unique($usedClasses);
+        $usedClassesSection = $this->formatUsedClasses(
+            $model->getBaseNamespace(),
+            $usedClasses
+        );
+        $template = str_replace('{{imports}}', $usedClassesSection, $template);
 
         return $template;
     }
 
+    /**
+     * Returns imports section for model.
+     *
+     * @param string $baseNamespace base namespace to avoid importing classes from same namespace
+     * @param array $usedClasses Array of used in model classes
+     *
+     * @return string
+     */
+    private function formatUsedClasses($baseNamespace, $usedClasses)
+    {
+        $result = [];
+        foreach ($usedClasses as $usedClass) {
+            // Do not import classes from same namespace
+            $namespacePattern = str_replace('\\', '\\\\', "/{$baseNamespace}\\[a-zA-Z0-9_]*/");
+            if (! preg_match($namespacePattern, $usedClass)) {
+                $result[] = "use {$usedClass};";
+            }
+        }
+
+        sort($result);
+
+        return implode("\n", $result);
+    }
+
+    /**
+     * Extract and replace fully-qualified class names from placeholder.
+     *
+     * @param string $placeholder Placeholder to extract class names from. Rewrites value to content without FQN
+     *
+     * @return array Extracted FQN
+     */
+    private function extractUsedClasses(&$placeholder)
+    {
+        $classNamespaceRegExp = '/([\\\\a-zA-Z0-9_]*\\\\[\\\\a-zA-Z0-9_]*)/';
```

Comment:
Are you assuming all classes can only have a depth of 2 namespaces like `\FisrtLevel\SecondLevel`?

Reply:
I'm assuming that a depth of namespaces is at least 2 because in other case no need to import. You can see on screenshot that namespace with depth of 4 was succefully imported

Reply:
Here is example of detected namespaces: https://regex101.com/r/cMOb9V/2

Reply:
Thanks for merging!